unity/Assets/Scripts/PhysicsRemoteFPSAgentController.cs

```diff
@@ -1609,165 +1612,209 @@ private Vector3[] GenerateArcPoints(Vector3 startingPoint, Vector3 origin, float
-                    errorMessage = "Teleport target out of scene bounds.";
-                    actionFinished(false);
-                    return;
-                }
+        // keep undocumented until float: rotation is added to Stochastic
+        // public void TeleportFull(
+        //     Vector3 position,
+        //     float rotation,
+        //     float horizon,
+        //     bool standing,
+        //     bool forceAction = false
+        // ) {
+        //     TeleportFull(
+        //         position: position,
+        //         rotation: new Vector3(0, rotation, 0),
+        //         horizon: horizon,
+        //         standing: standing,
+        //         forceAction: forceAction
+        //     );
+        // }
 
-                if (!isPositionOnGrid(targetTeleport)) {
-                    errorMessage = $"Target teleport position {targetTeleport.ToString("F6")} is not" +
-                        $" on the grid of size {gridSize}.";
-                    actionFinished(false);
-                    return;
-                }
+        // has to consider both the arm and standing
+        public void TeleportFull(
+            Vector3 position,
+            Vector3 rotation,
+            float horizon,
+            bool standing,
+            bool forceAction = false
+        ) {
+            // cache old values in case there's a failure
+            bool wasStanding = isStanding();
+            Vector3 oldPosition = transform.position;
+            Quaternion oldRotation = transform.rotation;
+            Vector3 oldCameraEulerAngle = m_Camera.transform.eulerAngles;
 
-                Vector3 oldPosition = transform.position;
-                Quaternion oldRotation = transform.rotation;
-                Vector3 oldLocalHandPosition = new Vector3();
-                Quaternion oldLocalHandRotation = new Quaternion();
-                if (ItemInHand != null) {
-                    oldLocalHandPosition = ItemInHand.transform.localPosition;
-                    oldLocalHandRotation = ItemInHand.transform.localRotation;
-                }
-                Vector3 oldCameraLocalEulerAngle = m_Camera.transform.localEulerAngles;
-                Vector3 oldCameraLocalPosition = m_Camera.transform.localPosition;
+            Vector3 oldLocalHandPosition = new Vector3();
+            Quaternion oldLocalHandRotation = new Quaternion();
+            if (ItemInHand != null) {
+                oldLocalHandPosition = ItemInHand.transform.localPosition;
+                oldLocalHandRotation = ItemInHand.transform.localRotation;
+            }
 
-                //default high level hand when teleporting
+            try {
+                // default high level hand when teleporting
                 DefaultAgentHand();
+                ToggleArmColliders(arm: Arm, value: forceAction);
+                base.TeleportFull(position: position, rotation: rotation, horizon: horizon, forceAction: forceAction);
```

Comment:
Won't this call already potentially set `actionFinished(true)` and thus allow unity to emit the metadata? Perhaps you want to create a `teleportFull` method that can be called programmatically?